### PR TITLE
EEGLAB import: empty channel positions were ingnored, which could lead to a shift in channel positions

### PR DESCRIPTION
**Pull request by mmagnuski**
Although empty channel positions were turned to np.nan such channel positions were later ignored leading to a shift in channel positions (in my case for example the O2 channel moved to where EOG should be).

On master (look at the eog channels and the occipital channels):
image

on this PR:
image

I'll add a test tomorrow and whatsneew, I'm just poking the CIs now to see whether the check I removed was actually needed anywhere. 

Followup: should nan-position channels be plotted at 0, 0 when using `inst.plot_sensors()`?

### CODE REVIEW

**Comment by mmagnuski** [comment]
Tests are ok, apart from one weird error on linux / pipi / py3.9:
```
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  25 (X_SendEvent)
  Resource id in failed request:  0x400002
  Serial number of failed request:  71
  Current serial number in output stream:  71
```

**Comment by larsoner** [comment]
Ignore that, that test bombs sometimes. We can restart if this happens once in MRG state, ignore it in the meantime (the other runs should be sufficient)

**Review by agramfort — COMMENTED** [review]
good to go from my end one we have a test and some entry in what's new

thx @mmagnuski

**Comment by mmagnuski** [comment]
I'll finish this today, sorry for delays.

**Comment by hoechenberger** [comment]
@mmagnuski Care to finish this one up? :)

**Comment by hoechenberger** [comment]
> Followup: should nan-position channels be plotted at 0, 0 when using `inst.plot_sensors()`?

I think they shouldn't appear at all …

**Comment by mmagnuski** [comment]
@hoechenberger Yes, my "tomorrow" seems to have been an approximation of 1.5 month :/. But I finally have more time this and the following week so I'm going to be more active here (meaning: mne, not just this PR). :)

**Comment by mmagnuski** [comment]
Hi, I just got bit by it again, so I will remember to finish this now. :)

**Comment by hoechenberger** [comment]
So, "tomorrow", I hear you say? 🤣😅 Just kidding, it's great you're picking this one up again!!! 💪💪💪

**Comment by mmagnuski** [comment]
I didn't request the reviews, this is due to change in files from the catastrophic PR update. Is there anyway easy way to save this PR or is it just best to open a new one?

**Comment by hoechenberger** [comment]
I'd just open a new one instead of spending 20+ min trying to salvage this one :)

**Comment by sappelhoff** [comment]
this should probably work for fixing:

```
git checkout main
git pull upstream main
git checkout -b bak
git cherry-pick <cherry pick your commits that you actually want from fix_eeglab_chan_pos>
git checkout fix_eeglab_chan_pos
git reset --hard upstream/main
git cherry-pick <cherry pick the commits you put on "bak">
git push --force
<verify all is well>
git branch -D bak
```

**Comment by mmagnuski** [comment]
CI failures are due to timeouts on Azure.

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -130,6 +130,8 @@ Bugs
 
 - :func:`mne.read_evokeds`, :func:`mne.channels.read_custom_montage`, :func:`mne.channels.read_dig_hpts`, :func:`mne.channels.read_dig_polhemus_isotrak`, and :func:`mne.channels.read_polhemus_fastscan` now correctly expand ``~`` in the provided path to the user's home directory (:gh:`10685`, :gh:`10688` by `Richard Höchenberger`_)
 
+- Fix bug in :func:`mne.io.read_raw_eeglab` and :func:`mne.io.read_epochs_eeglab`, where the presence of channels without position in the .set file caused positions incorrectly assigned to all further channels (:gh:`8754` by `Mikołaj Magnuski`_)
```
Comment (hoechenberger) [inline doc/changes/latest.inc:133]:
```suggestion
- Fix bug in :func:`mne.io.read_raw_eeglab` and :func:`mne.io.read_epochs_eeglab`, where the presence of channels without position in the ``.set`` file caused positions incorrectly assigned to all further channels (:gh:`8754` by `Mikołaj Magnuski`_)
```

**Review by hoechenberger — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @mmagnuski !

**Comment by mmagnuski** [comment]
🚀

**Pull request merged**